src/core/lib/surface/channel.cc

```diff
@@ -143,10 +144,15 @@ static grpc_core::UniquePtr<char> get_default_authority(
     } else if (0 == strcmp(input_args->args[i].key,
                            GRPC_SSL_TARGET_NAME_OVERRIDE_ARG)) {
       ssl_override = grpc_channel_arg_get_string(&input_args->args[i]);
+    } else if (0 == strcmp(input_args->args[i].key,
+                           GRPC_ALTS_TARGET_NAME_OVERRIDE_ARG)) {
```

Comment:
I'm really not keen on adding this special case to the channel code itself.  Can we instead set this in the ALTS credentials?

I realize that the corresponding case for SSL creds is here, but I think that's a mistake.  I'd like to see that one moved into SSL creds as well.

Reply:
Done, I moved both the SSL and ALTS override codes into the channel_credentials [update_arguments](https://github.com/grpc/grpc/blob/83dd71dfe58253b4ceb3cf394a939f70adc7ea95/src/core/lib/security/credentials/credentials.h#L128) hook.

This then orphaned this function in `src/core/lib/surface/channel.cc`.